designdocs/Deterministic wallets.txt

```diff
@@ -1,9 +1,15 @@
-Design of deterministic wallets support
----------------------------------------
+Design of hierarchical deterministic (HD) wallets support
+---------------------------------------------------------
 
 Goals:
-- Wallets to derive new keys deterministically using the BIP32 algorithm.
-- Seamless/silent upgrade of old wallets to use the new scheme
+- Wallets to derive new keys deterministically using the BIP32 algorithm
+- Support different wallet structures:
+  - The default BIP32 structure, described in the later part of the document
+  - BIP43/44/84/86
+  - Custom, by implementing the KeyChainGroupStructure interface
+- Support multiple output script types (P2PKH, P2WPKH)
+  - One active chain per script type, so that compatible addresses can be generated
```

Comment:
Should say one active "keychain" (I interpreted that as blockchain on first read and was momentarily confused)

Reply:
Good point. I added a commit that uses the term 'keychain' everywhere in the document.